app/actions/remote/post.ts

```diff
@@ -26,6 +26,83 @@ type AuthorsRequest = {
     error?: unknown;
 }
 
+export const createPost = async (serverUrl: string, post: Partial<Post>, files: FileInfo[] = []): Promise<{data?: boolean; error?: any}> => {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    const currentUserId = queryCurrentUserId(operator.database);
+    const timestamp = Date.now();
+    const pendingPostId = post.pending_post_id || `${currentUserId}:${timestamp}`;
+
+    // Check if is sending this post id
+
+    let newPost = {
+        ...post,
+        id: '',
+        pending_post_id: pendingPostId,
+        create_at: timestamp,
+        update_at: timestamp,
+        ownPost: true,
+    } as Post;
+
+    if (files.length) {
+        const fileIds = files.map((file) => file.id);
+
+        newPost = {
+            ...newPost,
+            file_ids: fileIds,
+        };
+    }
+
+    const databasePost = {
+        ...newPost,
+        id: pendingPostId,
+    };
+
+    await operator.handlePosts({
+        actionType: ActionType.POSTS.RECEIVED_NEW,
+        order: [databasePost.id],
+        posts: [databasePost],
+    });
+
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const created = await client.createPost(newPost);
+        await operator.handlePosts({
+            actionType: ActionType.POSTS.RECEIVED_NEW,
+            order: [created.id],
+            posts: [created],
+        });
+        newPost = created;
+    } catch (error: any) {
+        const errorPost = {
+            ...newPost,
+            id: pendingPostId,
+            failed: true,
+            update_at: Date.now(),
+        };
+
+        // If the failure was because: the root post was deleted or
+        // TownSquareIsReadOnly=true then remove the post
+        if (error.server_error_id === 'api.post.create_post.root_id.app_error' ||
+            error.server_error_id === 'api.post.create_post.town_square_read_only' ||
+            error.server_error_id === 'plugin.message_will_be_posted.dismiss_post'
+        ) {
+            // Delete Post
```

Comment:
are you going to add the delete post code in this PR?

Reply:
Yes, I missed that.